src/commands/org/delete/shape.ts

```diff
@@ -0,0 +1,164 @@
+  public static readonly deprecateAliases = true;
+
+  public static readonly flags = {
+    'target-org': requiredOrgFlagWithDeprecations,
+    'api-version': orgApiVersionFlagWithDeprecations,
+    loglevel,
+    'no-prompt': Flags.boolean({
+      char: 'p',
+      summary: messages.getMessage('noPrompt'),
+      description: messages.getMessage('noPromptLong'),
+      aliases: ['noprompt'],
+      deprecateAliases: true,
+    }),
+  };
+
+  public async run(): Promise<OrgShapeDeleteResult> {
+    const { flags } = await this.parse(OrgShapeDeleteCommand);
+    const username = flags['target-org'].getUsername();
+    const orgId = flags['target-org'].getOrgId();
+    if (!flags['no-prompt']) {
+      if (!(await this.confirm(messages.getMessage('deleteCommandYesNo', [username])))) {
+        return;
+      }
+    }
+
+    const conn = flags['target-org'].getConnection(flags['api-version']);
+
+    if (!(await isShapeEnabled(conn))) {
+      throw messages.createError('noAccess', [username]);
+    }
+
+    const deleteRes = await deleteAll(conn, username);
+
+    if (deleteRes.shapeIds.length === 0) {
+      this.log(messages.getMessage('noShapesHumanSuccess', [orgId]));
+      return;
+    }
+
+    if (deleteRes.failures.length > 0 && deleteRes.shapeIds.length > 0) {
+      setExitCode(68);
+
+      this.styledHeader('Partial Success');
+      this.log(messages.getMessage('humanSuccess', [orgId]));
+      this.log('');
+      this.styledHeader('Failures');
+      const columns = {
+        shapeId: { header: 'Shape ID' },
+        message: { header: 'Error Message' },
+      };
+      this.table(deleteRes.failures, columns);
+    } else if (deleteRes.failures.length > 0) {
+      setExitCode(1);
+    } else if (deleteRes.shapeIds.length > 0) {
+      setExitCode(0);
+      this.log(messages.getMessage('humanSuccess', [orgId]));
+    }
+
+    return {
+      orgId,
+      shapeIds: deleteRes.shapeIds,
+      failures: deleteRes.failures,
+    };
+  }
+}
+
+/**
+ * Delete all ShapeRepresentation records for the shapeOrg.
+ *
+ * @return List of SR IDs that were deleted
+ */
+export const deleteAll = async (conn: Connection, username: string): Promise<DeleteAllResult> => {
+  const deleteAllResult = {
+    shapeIds: [],
+    failures: [],
+  };
+
+  let shapeIds: string[] = [];
+
+  try {
+    const result = await conn.query<ShapeRepresentation>('SELECT Id FROM ShapeRepresentation');
+    if (result.totalSize === 0) {
+      return deleteAllResult;
+    }
+    shapeIds = result.records.map((shape) => shape.Id);
+  } catch (err) {
+    const JsForceErr = err as JsForceError;
+    if (JsForceErr.errorCode && JsForceErr.errorCode === 'INVALID_TYPE') {
+      // ShapeExportPref is not enabled, or user does not have CRUD access
+      throw messages.createError('noAccess', [username]);
+    }
+    // non-access error
+    throw JsForceErr;
+  }
+
+  await Promise.allSettled(
```

Comment:
I have been thinking about this and I can't figure out why `allSettled` is needed here. Each promise produced uses a try catch, so there no way a promise with fail.

Seems that Promise.all is sufficient.

How it could be used would be
* The success would simply return the shapeId successfully deleted
*  the catch capturing the shapeId and error message and then throw an error that contains those two bits in the error

Then one would post process the results of the `allSettled` to extract the reportable values from the settled promises.

Reply:
agree.  I went with `.all` because it was the least change.

FWIW, the TS support for narrowing allSettled is kinda ugly.